spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/EnableRabbitIntegrationTests.java

```diff
@@ -1005,6 +1057,36 @@ public Foo2 foo2(Foo2 foo2) {
 			return foo2;
 		}
 
+		@RabbitListener(queues="test.converted.list")
+		public Foo2 foo2(List<Foo2> foo2s) {
+			Foo2 foo2 = foo2s.get(0);
+			foo2.setBar("BAZZZZ");
+			return foo2;
+		}
+
+		@RabbitListener(queues="test.converted.array")
+		public Foo2 foo2(Foo2[] foo2s) {
+			Foo2 foo2 = foo2s[0];
+			foo2.setBar("BAZZxx");
+			return foo2;
+		}
+
+		@RabbitListener(queues="test.converted.args1")
+		public String foo2(Foo2 foo2, @Header("amqp_consumerQueue") String queue) {
+			return foo2 + queue;
+		}
+
+		@RabbitListener(queues="test.converted.args2")
+		public String foo2a(@Payload Foo2 foo2, @Header("amqp_consumerQueue") String queue) {
+			return foo2 + queue;
+		}
+
+		@RabbitListener(queues="test.converted.message")
+		public String foo2Message(@Payload Foo2 foo2, Message message) {
```

Comment:
OK. As I expected:
```
@RabbitListener(queues="test.notconverted.message")
public String justMessage(Message message) {
	return "foo";
}
```

ends up with the:
```
com.fasterxml.jackson.databind.JsonMappingException: No suitable constructor found for type [simple type, class org.springframework.amqp.core.Message]: can not instantiate from JSON object (missing default constructor or creator, or perhaps need to add/enable type information?)
 at [Source: { "bar" : "baz" }; line: 1, column: 3]
```
Fixing the single param inferring as I commented above makes me happy for the test like:
```
returned = template.convertSendAndReceive("", "test.notconverted.message", "{ \"bar\" : \"baz\" }",
				messagePostProcessor);
```
with the result:
```
"(Body:'{ \"bar\" : \"baz\" }' MessageProperties [headers={}, timestamp=null, messageId=null, userId=null, appId=null, clusterId=null, type=null, correlationId=null, replyTo=amq.rabbitmq.reply-to.g2dkAAtyYWJiaXRASE9NRQAACPEAAAAAAw==.DwnvYpsCuMKIeOumJkRO9w==, contentType=application/json, contentEncoding=UTF-8, contentLength=0, deliveryMode=null, expiration=null, priority=0, redelivered=false, receivedExchange=, receivedRoutingKey=test.notconverted.message, deliveryTag=1, messageCount=0])justMessageFoo2Service"
```